src/current/v23.1/goldengate.md

```diff
@@ -0,0 +1,458 @@
+
+    ~~~
+    # This is needed so that all the Postgres libraries can be found
+    export LD_LIBRARY_PATH=/usr/pgsql-13/lib:/u01/ggs-pg/lib
+
+    # This is needed so that OGG knows where to look for connection details for the database
+    export ODBCINI=/etc/odbc.ini
+
+    vi /etc/odbc.ini
+
+    # Inside put this: replace the details for log in with your own. Be sure to prefix the database name with {hostname}.{db}
+    # No changes should be needed for CRDBLOCAL
+    [ODBC Data Sources]
+    PG_src=DataDirect 7.1 PostgreSQL Wire Protocol
+    PG_tgt=DataDirect 7.1 PostgreSQL Wire Protocol
+    CRDB=DataDirect 7.1 PostgreSQL Wire Protocol
+
+    [ODBC]
+    IANAAppCodePage=4
+    InstallDir=/u01/ggs-pg
+
+
+    [CRDBSERVERLESS]
+    Driver=/u01/ggs-pg/lib/GGpsql25.so
+    Description=DataDirect 7.1 PostgreSQL Wire Protocol
+    Database={host name}.defaultdb
+    HostName={host}
+    PortNumber=26257
+    LogonID={your sql user}
+    Password={your sql user's pass}
+    EncryptionMethod=1 
+    ValidateServerCertificate=1 
+    TrustStore=/root/.postgresql/root.crt
+    ~~~
+
+1. Ensure that all the PostgreSQL libraries are installed and referenced in the `LD_LIBRARY_PATH`. 
+    
+    The path should at least include `/usr/pgsql-13/lib` and `/u01/ggs-pg/lib`.
+
+1. Log in to the database:
+
+    ~~~
+    cd /u01/ggs-pg
+    export LD_LIBRARY_PATH=/usr/pgsql-13/lib:/u01/ggs-pg/lib
+    export ODBCINI=/etc/odbc.ini
+
+    ./ggsci
+
+    # To log into the local database
+    DBLOGIN SOURCEDB CRDBLOCAL
+
+    # To log into the serverless database and then enter your password
+    DBLOGIN SOURCEDB CRDBSERVERLESS
+    ~~~
+
+## Set up Extract
+
+[Extract](https://docs.oracle.com/goldengate/c1230/gg-winux/GGCON/processes-and-terminology.htm) is Oracle GoldenGate's data capture mechanism that is configured to run against the source database.
+
+1. In a terminal:
+
+    ~~~
+    # Setup the environment variables for the Oracle source, which includes the ORACLE_HOME, TNS_ADMIN, LD_LIBRARY_PATH, and ORACLE_SID
+
+    cd /u01/ggs # Otherwise known as $OGG_HOME
+    ./ggsci
+
+    # Inside the ggsci terminal run these
+    edit param epos
+
+    view param epos
+
+    # This is the resulting output
+    # This section will run an extract on the source Oracle DB and then send that data over to the trail file at `./dirdat/ab` on the remote host.
+    EXTRACT epos
+    USERIDALIAS gg_source
+    RMTHOST {host-name}, MGRPORT {port}
+    RMTTRAIL ./dirdat/ab
+    TABLE OGGADM1.testtable;
+
+
+    # Back in the ./ggsci terminal
+    add extract epos, tranlog, begin now
+    add rmttrail ./dirdat/ab, extract epos megabytes 10
+
+    # This is crucial in order to address the logmining not found error
+    register EXTRACT epos, DATABASE
+
+    # Starts service
+    start epos
+    ~~~
+
+1. Check the status of the Extract by adding a row to `oggadm1.testtable`, committing it to the source, and running the following:
```

Comment:
Make sure to note that this is on the machine and in the directory that `Oracle Golden Gate for Oracle` is installed.